src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.SkipValue.cs

```diff
@@ -2,28 +2,50 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+using System.Diagnostics;
+
 namespace System.Security.Cryptography.Encoding.Tests.Cbor
 {
     internal partial class CborReader
     {
-        public void SkipValue()
+        // flag used to temporarily disable conformance level checks,
+        // e.g. during a skip operation over nonconforming encodings.
+        private bool _isConformanceLevelCheckEnabled = true;
+
+        public void SkipValue(bool validateConformance = false) => SkipToAncestor(0, validateConformance);
+        public void SkipToParent(bool validateConformance = false)
         {
-            CborReaderCheckpoint checkpoint = CreateCheckpoint();
+            if ((_nestedDataItems?.Count ?? 0) == 0)
+            {
+                throw new InvalidOperationException("CBOR reader is at the root context.");
+            }
+
+            SkipToAncestor(1, validateConformance);
+        }
+
+        private void SkipToAncestor(int depth, bool validateConformance)
```

Comment:
Could possibly make this method public, but I suspect it might confuse users. Similar results can be achieved with repeated applications of SkipToParent().

Reply:
This might actual be needed for skipping malformed data in COSE key parsing.

Reply:
I'm not sure that I follow what the scenario would be. I agree with the notion that it's just repeated calls to SkipToParent, and don't know it's necessary to expose the version that takes a count.

Reply:
From RFC8152:

> Each element in a COSE Key Set MUST be processed independently.  If
   one element in a COSE Key Set is either malformed or uses a key that
   is not understood by an application, that key is ignored and the
   other keys are processed normally.

Basically, assuming that a cose key set is defined at depth _d_, and given a nested cose key where a format error occurs at depth _d'_, where _d < d'_, we'd need to skip _d' - d_ levels. It feels like exposing an (appropriately named) skip method that takes a depth (or is it height?) parameter might save us a few cpu cycles.

Reply:
Eh, it might end up being a loss after the argument validation :smile:.  Since it's already possible to do (and not very complicated) I'd hold this until there's a request for it.